doc/src/guides/solving/solve-matrix-equation.md

```diff
@@ -0,0 +1,124 @@
+# Solve a Matrix Equation
+
+Use SymPy to solve a matrix (linear) equation. For example, solving $
+\left[\begin{array}{cc} c & d\\1 & -e\end{array}\right] \left[\begin{array}{cc}
+x\\y\end{array}\right] = \left[\begin{array}{cc} 2\\0\end{array}\right] $ yields
+$ \left[\begin{array}{cc} x\\y\end{array}\right] = \left[\begin{array}{cc}
+\frac{2e}{ce+d}\\\frac{2}{ce+d}\end{array}\right]$.
+
+## Alternatives to consider
+- Solving a matrix equation is equivalent to solving a system of linear
+  equations, so if you prefer you can use *link to solve a system of equations
+  once published*
+- *alternative 2*
+
+## Solve a Matrix Equation
+
+Here is an example of solving a matrix equation with SymPy's
+{meth}`sympy.matrices.matrices.MatrixBase.solve`. We use the standard matrix
+equation formulation $Ax=b$ where
+- $A$ is the matrix representing the coefficients in the linear equations
+- $x$ is the column vector of unknowns to be solved for
+- $b$ is the column vector of constants, where each row is the value of an
+  equation
+
+```py
+>>> from sympy import symbols
+>>> from sympy.matrices import Matrix
+>>> c, d, e = symbols("c, d, e")
+>>> A = Matrix([[c,d], [1, -e]])
+>>> A
+Matrix([
+[c,  d],
+[1, -e]])
+>>> b = Matrix([2, 0])
+>>> b
+Matrix([
+[2],
+[0]])
+>>> A.solve(b)
+Matrix([
+[2*e/(c*e + d)],
+[  2/(c*e + d)]])
+```
+
+## Guidance
+
+### Matrix Must Be Square
+
+The matrix $A$ must be square to represent a system of linear equations with the
+same number of unknowns as equations. If not, SymPy will give an error:
+
+```py
+ShapeError: `self` and `rhs` must have the same number of rows.
+```
+
+### *Guidance 2*
+
+*Guidance 2 content*
+
+
+## *Title*
+
+You can *title* in several ways. 
+
+### *Method 1*
+
+*Method 1 content*
+
+### *Method 2*
+
+*Method 2 content*
+
+## Working With Symbolic Matrices
```

Comment:
@asmeurer, as you suggested, requesting that you and @oscarbenjamin please provide input on caveats for working with symbolic, as opposed to numerical, matrices.

Reply:
Notes on difficulties for working with symbolic matrices from @asmeurer to include in page:
In numeric matrices, have to worry about numerical stability, condition numbers. Those don't matter for symbolic matrices; what matters is that SymPy's algorithms (which are not at all the same as with numeric linear algebra library) can greatly increase complexity of expressions in matrix. Sort of problems you can solve with symbolic matrices is more limited because of that. For example, if every entry in a matrix is a symbol, SymPy may take a long time to solve because size of symbolic determinant increases with factorial of matrix size.

Examples
- 4x4 matrix, MatrixSymbol('A', 4, 4).as_explicit().det() -- vary n from 3 up until it takes way too long to solve. (Solving a matrix equation involves computing its inverse, thus its determinant--per Kramer's Rule.)
- Reduced-row echelon form can produce an expression with a lot terms

The more unrelated, symbolic entries, the more likely you are to run into these problems.

How to address such problems: Only general tip is to use [domain matrix class](https://docs.sympy.org/latest/modules/polys/domainmatrix.html?highlight=domain+matrix#sympy.polys.matrices.domainmatrix.DomainMatrix)--can be faster because limits the domain of matrix entries.

Any other points, @oscarbenjamin?